### PR TITLE
New memory allocator and initializer for vectors and matrices 

### PR DESCRIPTION
**Pull request by NaveenElumalaiAMD**
resolves: [LWPMLSE-642](https://ontrack-internal.amd.com/browse/LWPMLSE-642)

### CODE REVIEW

File: clients/include/blas2/testing_spr_batched.hpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ void testing_spr_batched_bad_arg(const Arguments& arg)
         int64_t N           = 100;
         int64_t incx        = 1;
         int64_t batch_count = 2;
-        int64_t A_size      = N * (N + 1) / 2;
+        int64_t size_A      = N * (N + 1) / 2;
```
Comment (TorreZuk) [inline clients/include/blas2/testing_spr_batched.hpp]: hipblas_packed_matrix_size(N);   
if you were trying to hit them all?

File: clients/include/blas2/testing_spr_batched.hpp [hunk 2]
```diff
@@ -147,7 +148,7 @@ void testing_spr_batched(const Arguments& arg)
     int64_t           batch_count = arg.batch_count;
 
     int64_t abs_incx = incx < 0 ? -incx : incx;
-    size_t  A_size   = size_t(N) * (N + 1) / 2;
+    size_t  size_A   = size_t(N) * (N + 1) / 2;
```
Comment (TorreZuk) [inline clients/include/blas2/testing_spr_batched.hpp]: hipblas_packed_matrix_size(N);

File: clients/include/blas2/testing_spr_batched.hpp [hunk 3]
```diff
@@ -166,76 +167,102 @@ void testing_spr_batched(const Arguments& arg)
...
-            hipblasSprBatchedFn,
-            (handle, uplo, N, &h_alpha, dx.ptr_on_device(), incx, dA.ptr_on_device(), batch_count));
-
-        CHECK_HIP_ERROR(hA_host.transfer_from(dA));
-        CHECK_HIP_ERROR(dA.transfer_from(hA));
+        DAPI_CHECK(hipblasSprBatchedFn,
+                   (handle,
+                    uplo,
+                    N,
+                    &h_alpha,
```
Comment (TorreZuk) [inline clients/include/blas2/testing_spr_batched.hpp:225]: This just halpha?

**Review by TorreZuk — APPROVED** [review]
Very good and nice to have it match rocBLAS.  Probably you should manually print some matrices to ensure the initialization ranges haven't changed.  Might be another test to add at some point.

File: clients/include/blas2/testing_spr_batched.hpp (hunk 3 above)
Comment (NaveenElumalaiAMD) [reply clients/include/blas2/testing_spr_batched.hpp:225]: `h_alpha` is not a vector it contains the value of alpha. But `halpha` is a vector.

**Pull request merged**